src/deno_transport.ts

```diff
@@ -93,14 +93,13 @@ export class DenoTransport implements Transport {
     const inbound = new DataBuffer();
     let pm: string;
     while (true) {
-      let c = await this.conn.read(this.buf);
-      if (c) {
-        if (c === null) {
-          // EOF
-          return Promise.reject(
-            new Error("socket closed while expecting INFO"),
-          );
-        }
+      const c = await this.conn.read(this.buf);
+      if (c === null) {
+        // EOF
+        return Promise.reject(
+          new Error("socket closed while expecting INFO"),
+        );
+      } else if (c) {
```

Comment:
Nit: if we are in a `else` of `if (c === null)` what is the purpose of `if (c)`? would a simple `else` be enough? If not, what other value can `c` have and what are we doing with that condition?

Reply:
The check for null is special, because it API call can return number or null. Null means EOF. with the check for (c), if it is greater than zero, we process otherwise we ignore.